### ISSUE TITLE
ToC animation for headings with umlauts not working

### ISSUE THREAD

**Issue by evolu1** [yzhang-gh/vscode-markdown#1109]
## Problem

When clicking a heading in a Table of Contents that contains umlauts/Umlaute (`ä`, `ö`, `ü`), the animation jumping down to the content is not smooth and only jumps near the content with a certain offset.  
This may be the case for other symbols as well. Not tested.

## How to reproduce

- .md file with headings containing umlauts/Umlaute (`ä`, `ö`, `ü`)   
(and content under the headings to have space for scrolling)
- adding sections numbers
- creating ToC
- print to html
- opening html and clicking the headings

## Environment

version 3.4.0 (2020.11.14)

**Comment by yzhang-gh** [comment]
It works fine on my machine. (And it actually has nothing to do with this extension.)

I guess that is because your document is not long enough (so it cannot scroll to the position).

**Comment by evolu1** [comment]
> It works fine on my machine. (And it actually has nothing to do with this extension.)

**With some testing now I can confirm that. Thank you!**
The navigation animation works with umlauts, when just using the generated html.

However, I use the html code that vscode-markdown generates in a .php file in Wordpress. (Full and complete html code used, not edited).
There the animation when clicking a heading is slightly different and for umlauts it is broken.
So the problem is my environment, Wordpress etc.
Any idea how I can find out what is causing this? 
Is this animation done with CSS? 
How can I overwrite the animation that is coming from the environment so that the animation in the generated html is being used? (like `!important` does for a single line).

**Comment by evolu1** [comment]
Thanks for the help! 
I checked that and in the end found out, it is an animation coming from some external javascript on the page.
Seemingly the page cannot handle links containing ä, ö, ü. 
Since I need a fix for this issue I looked for ways to replace the umlauts in the links.

**Quick fix**

I realized the links in vscode-markdown get generated by slugify. I added a few lines in `/src/toc.ts` before the string gets slugify'ed in order to replace the umlauts for me. After line 668:
```
const anchorOccurrences = new Map<string, number>();
function getSlug(rawContent: string): string {
    //replace umlauts START
    let re = /ä/gi;
    rawContent = rawContent.replace(re, "ae");
    let ree = /ö/gi;
    rawContent = rawContent.replace(ree, "oe");
    let reee = /ü/gi;
    rawContent = rawContent.replace(reee, "ue");
    //replace umlauts STOP
    let slug = slugify(rawContent, { env, mode: slugifyMode });
```
I then compiled the extension and installed it locally.
Now, more umlauts in internal links 👍 

For other users who run into this issue:
- I cloned the repository, changed the code
- `npm install`
- `npm install -g vsce`
- `vsce package` which compiles the .vsix file 
- install .vsix via VSC extension manager

/closed
